### PR TITLE
Add default option - fixes #10

### LINKED ISSUE #10: Override `null` fallback
It's a minor feature, but I'd like to be able to pass an option to set the default `null` result to something else. This could be handy when you want to fall back on `true`, e.g. `yn(req.query.something, { default: true })`.

### PR DESCRIPTION
**Pull request by SamVerschueren**
Feel free to provide feedback. I also added an `API` section to the documentation.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -39,15 +39,41 @@ yn(true);
 yn('abomasum');
 //=> null
 
-// lenient mode will use a key distance-based score
-// to leniently accept typos of "yes" and "no"
```
Comment (SamVerschueren) [inline readme.md]: Moved this to the `API` section

File: readme.md [hunk 2]
```diff
@@ -39,15 +39,41 @@ yn(true);
...
 Unrecognized values return `null`.
 
 
+## API
+
+### yn(input, [options])
+
+#### input
+
+Type: `*`
```
Comment (SamVerschueren) [inline readme.md]: Not sure about the type...

File: index.js [hunk 3]
```diff
@@ -1,9 +1,17 @@
...
 
 module.exports = function (val, opts) {
 	val = String(val).trim();
-	opts = opts || {};
+	opts = objectAssign({
+		lenient: false,
+		default: null
+	}, opts);
+
+	if (opts.default !== null && typeof opts.default !== 'boolean') {
```
Comment (SamVerschueren) [inline index.js]: Not sure if you want to enable every kind of value as a `default`?

File: readme.md (hunk 2 above)
Comment (sindresorhus) [inline readme.md]: I usually use `any`, like Flow.

File: readme.md [hunk 4]
```diff
@@ -39,15 +39,43 @@ yn(true);
...
+
+Value that should be converted.
+
+#### options
+
```
Comment (sindresorhus) [inline readme.md]: Add

```
Type: `Object`
```

For completeness.

File: index.js (hunk 3 above)
Comment (sindresorhus) [inline index.js]: Only boolean.

File: index.js [hunk 5]
```diff
@@ -1,9 +1,17 @@
 'use strict';
+var objectAssign = require('object-assign');
```
Comment (sindresorhus) [inline index.js]: Yuk. We should really ES2015ify after this.

**Comment by sindresorhus** [comment]
// @blakeembrey

**Comment by blakeembrey** [comment]
Thanks. Looks like what I was after.

Not sure if you care about the `default` restriction though, couldn't you leave it open-ended? The only thing I can think of is defaulting `undefined` in some places, but I'm good with just boolean to start either way.

**Comment by sindresorhus** [comment]
> Not sure if you care about the default restriction though, couldn't you leave it open-ended?

I prefer being restrictive and rather loosen it if genuine use-cases pops up.

File: index.js (hunk 5 above)
Comment (SamVerschueren) [inline index.js]: Definitely! I will get to it once this one is merged and released.

**Comment by SamVerschueren** [comment]
Processed feedback

**Comment by sindresorhus** [comment]
🙌

**Pull request merged**